### PR TITLE
Properly inject missing version/date/commit variables

### PR DESCRIPTION
**Pull request by MarioUhrik**
Adding the missing "ldflags" section fixes the following issues:

Fixes https://github.com/controlplaneio/kubesec/issues/250
Fixes https://github.com/controlplaneio/kubesec/issues/337
FYI @06kellyjac 

I've validated this fix locally:

```
$ goreleaser build --snapshot --clean
  • starting build...
  • loading config file                              file=.goreleaser.yml
  • loading environment variables
  • getting and validating git state
    • ignoring errors because this is a snapshot     error=git doesn't contain any tags. Either add a tag or use --snapshot
    • building...                                    commit=5fc3bd266ec4e88f121f3ac168ad88ec9dcd895a latest tag=v0.0.0
    • pipe skipped                                   reason=disabled during snapshot mode
  • parsing tag
  • setting defaults
  • running before hooks
    • running                                        hook=go mod tidy
  • snapshotting
    • building snapshot...                           version=v0.0.0-next
  • checking distribution directory
    • cleaning dist
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/kubesec_linux_arm_6/kubesec
    • building                                       binary=dist/kubesec_darwin_arm64/kubesec
    • building                                       binary=dist/kubesec_darwin_amd64_v1/kubesec
    • building                                       binary=dist/kubesec_linux_amd64_v1/kubesec
    • building                                       binary=dist/kubesec_linux_arm_7/kubesec
    • building                                       binary=dist/kubesec_linux_arm64/kubesec
^R
    • took: 2s
  • storing release metadata
    • writing                                        file=dist/artifacts.json
    • writing                                        file=dist/metadata.json
  • build succeeded after 2s
  • thanks for using goreleaser!
$ ./dist/kubesec_linux_amd64_v1/kubesec version
version v0.0.0-next
git commit 5fc3bd266ec4e88f121f3ac168ad88ec9dcd895a
build date 2023-07-03T12:47:42Z
```

### CODE REVIEW

File: .goreleaser.yml
Comment (spiarh) [inline .goreleaser.yml:24]: `.Commit` is deprecated in favor of `.FullCommit` as per doc: https://goreleaser.com/customization/templates/

Comment (spiarh) [inline .goreleaser.yml:24]: `.Commit` is deprecated in favor of `.FullCommit` as per doc: https://goreleaser.com/customization/templates/

**Review by spiarh — DISMISSED** [review]
@MarioUhrik  Thanks for the contribution, only a minor change before merging

**Comment by MarioUhrik** [comment]
Hi @spiarh , thanks for the feedback.

I've done as you suggested.
It still works:
```
$ ./kubesec_linux_amd64_v1/kubesec version
version v0.0.0-next
git commit 7c879dab12fda048ebdadbb5ee90f06e114e139b]
build date 2023-07-19T14:31:13Z
```

**Comment by MarioUhrik** [comment]
I've also split the long line into 3 lines in order to pass the YAML linter, should be fine now.
Waiting for feedback

File: .goreleaser.yml
Comment (06kellyjac) [inline .goreleaser.yml:26]: `.Date` is not reproducible, we can use the date of the commit AKA `SOURCE_DATE_EPOCH` in RFC 3339 format instead

```suggestion
    ldflags:
      - -s -w
      - -X "github.com/controlplaneio/kubesec/v2/cmd.version={{.Version}}"
      - -X "github.com/controlplaneio/kubesec/v2/cmd.commit={{.FullCommit}}"
      - -X "github.com/controlplaneio/kubesec/v2/cmd.date={{.CommitDate}}"
```

**Review by 06kellyjac — CHANGES_REQUESTED** [review]
Thanks

Doesn't need `-s -w` repeated & other suggestions inline

**Comment by 06kellyjac** [comment]
Updated the description as this PR still wouldn't resolve https://github.com/controlplaneio/kubesec/issues/121 because the container builds do not use goreleaser

**Pull request merged**
